services/core/cards-publication/src/test/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessServiceShould.java

```diff
@@ -1107,4 +1108,48 @@ void processCardWithEntitiesRequiredToRespondNotSubsetOfEntitiesAllowedToRespond
                 "The number of registered cards should be '%d' but is " + "'%d' actually",
                 numberOfCards, block).isEqualTo(numberOfCards);
     }
+
+    @Test
+    void deleteCards_by_endDate() {
+
+        EasyRandom easyRandom = instantiateRandomCardGenerator();
+        int numberOfCards = 10;
+
+        List<CardPublicationData> cards = instantiateSeveralRandomCards(easyRandom, numberOfCards);
+        
+        Instant ref = Instant.now();
+        AtomicInteger i = new AtomicInteger(1);
+        cards.forEach(c -> {
```

Comment:
Why not using just int ?

Reply:
Variable must be final / effectively final to use it inside lambda.